src/tasks/LargeFileUploadTask.ts

```diff
@@ -146,10 +150,21 @@ export class LargeFileUploadTask {
 	 */
 	public constructor(client: Client, file: FileObject, uploadSession: LargeFileUploadSession, options: LargeFileUploadTaskOptions = {}) {
 		this.client = client;
-		this.file = file;
+
+		if (!file.sliceFile) {
+			console.warn("Please pass a fileObject instance");
+			try {
+				this.file = new FileUpload(file.content, file.name, file.size);
+			} catch (err) {
+				throw new Error("use file upload object");
```

Comment:
maybe pass this source error into the new error so the actual issue bubbles up?

Reply:
The code has changed. The outermost try-catch is removed. So the source error will be thrown at that line.